File: fixtures/nodejs/j02/fixture.js
// God class with all code smells — the model must refactor this
// Smells: god class, magic numbers, duplicated logic, deep nesting, long methods

class OrderProcessor {
  constructor(db, mailer, logger) {
    this.db = db;
    this.mailer = mailer;
    this.logger = logger;
    this.orders = [];
    this.customers = [];
    this.inventory = [];
    this.discounts = [];
    this.payments = [];
    this.shipments = [];
  }

  processOrder(order) {
    let discount = 0;
    if (order.items.length > 10) {
      if (order.customer.type === 'vip') {
        discount = 0.25;
      } else {
        discount = 0.1;
      }
    } else {
      if (order.customer.type === 'vip') {
        discount = 0.15;
      } else {
        discount = 0;
      }
    }

    let total = 0;
    for (let i = 0; i < order.items.length; i++) {
      const item = order.items[i];
      if (item.quantity > 0) {
        if (item.price > 0) {
          total += item.price * item.quantity;
        }
      }
    }
    total = total - (total * discount);

    if (total > 1000) {
      order.shipping = 0;
    } else if (total > 500) {
      order.shipping = 9.99;
    } else {
      order.shipping = 19.99;
    }
    total += order.shipping;

    if (order.paymentMethod === 'credit_card') {
      const fee = total * 0.029 + 0.30;
      total += fee;
    } else if (order.paymentMethod === 'paypal') {
      const fee = total * 0.034 + 0.30;
      total += fee;
    }

    order.total = total;
    order.status = 'processed';
    this.orders.push(order);

    // Send confirmation email (duplicated logic below in sendOrderUpdate)
    if (order.customer.email) {
      this.mailer.send({
        to: order.customer.email,
        subject: 'Order Confirmation',
        body: `Your order total is $${total.toFixed(2)}`,
      });
    }

    this.logger.log(`Order processed: ${JSON.stringify(order)}`);
    return order;
  }

  cancelOrder(orderId, reason) {
    const order = this.orders.find((o) => o.id === orderId);
    if (order) {
      if (order.status !== 'shipped') {
        if (order.status !== 'delivered') {
          order.status = 'cancelled';
          order.cancellationReason = reason;

          // Duplicated email sending logic
          if (order.customer.email) {
            this.mailer.send({
              to: order.customer.email,
              subject: 'Order Cancelled',
              body: `Your order has been cancelled. Reason: ${reason}`,
            });
          }

          let refundAmount = order.total;
          if (order.paymentMethod === 'credit_card') {
            refundAmount -= 0.30;
          }
          order.refundAmount = refundAmount;
          this.logger.log(`Order cancelled: ${orderId}`);
          return true;
        }
      }
    }
    return false;
  }

  sendOrderUpdate(orderId, updateType) {
    const order = this.orders.find((o) => o.id === orderId);
    if (order) {
      // Duplicated email sending logic (third time)
      if (order.customer.email) {
        this.mailer.send({
          to: order.customer.email,
          subject: `Order ${updateType}`,
          body: `Your order status: ${updateType}`,
        });
      }
      return true;
    }
    return false;
  }

  getInventoryStatus(productId) {
    const item = this.inventory.find((i) => i.id === productId);
    if (item) {
      if (item.quantity > 100) {
        return 'in_stock';
      } else if (item.quantity > 10) {
        return 'low_stock';
      } else if (item.quantity > 0) {
        return 'critical_stock';
      } else {
        return 'out_of_stock';
      }
    }
    return 'unknown';
  }
}

module.exports = { OrderProcessor };

File: fixtures/dotnet/j02/Fixture.cs
// God class with all code smells -- the model must refactor this
// Smells: god class, magic numbers, duplicated logic, deep nesting, long methods

public class OrderItem
{
    public double Price { get; set; }
    public int Quantity { get; set; }
}

public class Customer
{
    public string Type { get; set; } = "standard";
    public string? Email { get; set; }
}

public class Order
{
    public string Id { get; set; } = "";
    public Customer Customer { get; set; } = new();
    public List<OrderItem> Items { get; set; } = new();
    public string PaymentMethod { get; set; } = "cash";
    public double Shipping { get; set; }
    public double Total { get; set; }
    public string Status { get; set; } = "pending";
    public string? CancellationReason { get; set; }
    public double RefundAmount { get; set; }
}

public class InventoryItem
{
    public string Id { get; set; } = "";
    public int Quantity { get; set; }
}

public class EmailMessage
{
    public string To { get; set; } = "";
    public string Subject { get; set; } = "";
    public string Body { get; set; } = "";
}

public interface IMailer
{
    void Send(EmailMessage message);
}

public interface ILogger
{
    void Log(string message);
}

public class OrderProcessor
{
    private readonly object _db;
    private readonly IMailer _mailer;
    private readonly ILogger _logger;
    public List<Order> Orders { get; set; } = new();
    public List<Customer> Customers { get; set; } = new();
    public List<InventoryItem> Inventory { get; set; } = new();
    public List<object> Discounts { get; set; } = new();
    public List<object> Payments { get; set; } = new();
    public List<object> Shipments { get; set; } = new();

    public OrderProcessor(object db, IMailer mailer, ILogger logger)
    {
        _db = db;
        _mailer = mailer;
        _logger = logger;
    }

    public Order ProcessOrder(Order order)
    {
        double discount = 0;
        if (order.Items.Count > 10)
        {
            if (order.Customer.Type == "vip")
            {
                discount = 0.25;
            }
            else
            {
                discount = 0.1;
            }
        }
        else
        {
            if (order.Customer.Type == "vip")
            {
                discount = 0.15;
            }
            else
            {
                discount = 0;
            }
        }

        double total = 0;
        for (int i = 0; i < order.Items.Count; i++)
        {
            var item = order.Items[i];
            if (item.Quantity > 0)
            {
                if (item.Price > 0)
                {
                    total += item.Price * item.Quantity;
                }
            }
        }
        total = total - (total * discount);

        if (total > 1000)
        {
            order.Shipping = 0;
        }
        else if (total > 500)
        {
            order.Shipping = 9.99;
        }
        else
        {
            order.Shipping = 19.99;
        }
        total += order.Shipping;

        if (order.PaymentMethod == "credit_card")
        {
            var fee = total * 0.029 + 0.30;
            total += fee;
        }
        else if (order.PaymentMethod == "paypal")
        {
            var fee = total * 0.034 + 0.30;
            total += fee;
        }

        order.Total = total;
        order.Status = "processed";
        Orders.Add(order);

        // Send confirmation email (duplicated logic below in SendOrderUpdate)
        if (order.Customer.Email != null)
        {
            _mailer.Send(new EmailMessage
            {
                To = order.Customer.Email,
                Subject = "Order Confirmation",
                Body = $"Your order total is ${total:F2}"
            });
        }

        _logger.Log($"Order processed: {order.Id}");
        return order;
    }

    public bool CancelOrder(string orderId, string reason)
    {
        var order = Orders.FirstOrDefault(o => o.Id == orderId);
        if (order != null)
        {
            if (order.Status != "shipped")
            {
                if (order.Status != "delivered")
                {
                    order.Status = "cancelled";
                    order.CancellationReason = reason;

                    // Duplicated email sending logic
                    if (order.Customer.Email != null)
                    {
                        _mailer.Send(new EmailMessage
                        {
                            To = order.Customer.Email,
                            Subject = "Order Cancelled",
                            Body = $"Your order has been cancelled. Reason: {reason}"
                        });
                    }

                    double refundAmount = order.Total;
                    if (order.PaymentMethod == "credit_card")
                    {
                        refundAmount -= 0.30;
                    }
                    order.RefundAmount = refundAmount;
                    _logger.Log($"Order cancelled: {orderId}");
                    return true;
                }
            }
        }
        return false;
    }

    public bool SendOrderUpdate(string orderId, string updateType)
    {
        var order = Orders.FirstOrDefault(o => o.Id == orderId);
        if (order != null)
        {
            // Duplicated email sending logic (third time)
            if (order.Customer.Email != null)
            {
                _mailer.Send(new EmailMessage
                {
                    To = order.Customer.Email,
                    Subject = $"Order {updateType}",
                    Body = $"Your order status: {updateType}"
                });
            }
            return true;
        }
        return false;
    }

    public string GetInventoryStatus(string productId)
    {
        var item = Inventory.FirstOrDefault(i => i.Id == productId);
        if (item != null)
        {
            if (item.Quantity > 100)
            {
                return "in_stock";
            }
            else if (item.Quantity > 10)
            {
                return "low_stock";
            }
            else if (item.Quantity > 0)
            {
                return "critical_stock";
            }
            else
            {
                return "out_of_stock";
            }
        }
        return "unknown";
    }
}
